codacy/templates/fluentd/fluentd-conf.yaml

```diff
@@ -17,48 +17,63 @@ data:
     <filter **>
       @type record_modifier
       <record>
-        # This removes unneeded fields from the logs
         for_remove ${record["kubernetes"].delete("namespace_name"); record["kubernetes"].delete("pod_id"); record["kubernetes"].delete("namespace_labels"); record["kubernetes"].delete("container_info"); record["kubernetes"].delete("labels"); record.delete("container_info"); record.delete("docker")}
       </record>
+      # This removes unneeded fields from the logs
       remove_keys for_remove
     </filter>
 
+    # Move mested records to toplevel to be used as buffer key
+    <filter **>
+      @type record_modifier
+      <record>
+        pod_name ${record["kubernetes"]["pod_name"]}
+      </record>
+      <record>
+        container_name ${record["kubernetes"]["container_name"]}
+      </record>
+    </filter>
+
     # Output to S3
     <match **>
-      # docs: https://github.com/fluent/fluent-plugin-s3 and https://docs.fluentd.org/output/s3
+      # Docs: https://github.com/fluent/fluent-plugin-s3
       @type s3
       @log_level info
 
+      # Credentials
       aws_key_id {{ .Values.global.minio.accessKey }}
       aws_sec_key {{ .Values.global.minio.secretKey }}
 
+      # Bucket configuration
       s3_endpoint http://{{ .Values.global.minio.location }}:9000
-      force_path_style true # This prevents AWS SDK from breaking endpoint URL for minio
       s3_bucket {{ .Values.fluentdoperator.bucketName }}
+      force_path_style true # This prevents AWS SDK from breaking endpoint URL for minio
       auto_create_bucket true
 
-      #s3_object_key_format %{path}%Y/%m/%d/cluster-log-%{index}.%{file_extension}
-
-      path ${tag}/%Y/%m/%d/
-      s3_object_key_format %{path}%{index}.%{file_extension}
+      # Output location configuration
+      path ${container_name}/${pod_name}/%Y-%m-%d
+      s3_object_key_format %{path}_%{index}.%{file_extension}
 
       # We can't use this due to a bug in minio
       # <bucket_lifecycle_rule>
       #   id test-policy
       #   expiration_days {{ .Values.fluentdoperator.expirationDays }}
       # </bucket_lifecycle_rule>
 
+      store_as json
       <format>
         @type json
       </format>
 
-      <buffer tag,time>
+      # Check https://docs.fluentd.org/plugin-development/api-plugin-output
+      <buffer tag,time,container_name,pod_name>
         @type file
         path /var/log/fluentd-buffers/s3.buffer
-        timekey 600 # Time frames of 10 minutes
-        timekey_wait 0 # We want to send the logs from the timeframe to s3 as soon as possible
+        timekey 6h
+        timekey_wait 0s
         timekey_use_utc true
-        chunk_limit_size 256m
+        flush_interval 0s
```

Comment:
Can you explain this change plz?

Reply:
Is the correct way to say for it to send stuff to s3 in a time interval. 

Check the link above it has more details. 

We could use type immediately but thought this was was easier to change and having the same output

Reply:
Ok, from what I understand these are the conditions to flush with `flush_mode interval`:

```
1. its size reaches to the limit of bytesize or number of records
2. its timekey is expired and timekey_wait passes
3. it lives longer than flush_interval
```

Wasn't the bytesize given by the "chunk_limit_size" ?

The `timekey=6h` and `timekey_wait=0` means it is flushed right away after 6 hours. Is this right?

The `flush_interval` set to zero means that it flushes once it receives, right? Like @DReigada said, this might overwhelm minio.

Reply:
Read the new huge comment I put there. @afsmeira detected some weird behaviour and we went deeper in testing. We tried our best to explain there our findings.

If you think there is other way let us know.